src/otx/core/model/entity/base.py

```diff
@@ -206,3 +210,63 @@ def _reset_prediction_layer(self, num_classes: int) -> None:
             num_classes: Number of classes
         """
         raise NotImplementedError
+
+
+class OVModel(OTXModel, Generic[T_OTXBatchDataEntity, T_OTXBatchPredEntity]):
+    """Base class for the OpenVINO model."""
+
+    def __init__(self, num_classes: int, config: DictConfig) -> None:
+        config = inplace_num_classes(cfg=config, num_classes=num_classes)
+        self.model_name = config.pop("model_name")
```

Comment:
These optional config members look like a workaround. I suggest conducting a refactoring in the future and make them regular

Reply:
Actually, if model_name is not in config -> the error will be raised. I agree with you that we may refactor config parameters and maybe add some of them in constructor directly